include/kernels/gauge_loop_trace.cuh

```diff
@@ -0,0 +1,94 @@
+#pragma once
+
+#include <gauge_field_order.h>
+#include <quda_matrix.h>
+#include <index_helper.cuh>
+#include <kernel.h>
+#include <thread_array.h>
+#include <array.h>
+#include <reduce_helper.h>
+#include <reduction_kernel.h>
+#include <gauge_path_helper.cuh>
+
+namespace quda {
+
+  /**
+    @brief Return the batch block size used for multi reductions.
+  */
+  constexpr unsigned int max_n_batch_block_loop_trace() { return 8; }
+
+  template <typename store_t, int nColor_, QudaReconstructType recon_>
+  struct GaugeLoopTraceArg : public ReduceArg<array<double, 2>>  {
+    using real = typename mapper<store_t>::type;
+    using reduce_t = array<double, 2>;
+    static constexpr unsigned int max_n_batch_block = max_n_batch_block_loop_trace();
+    static constexpr int nColor = nColor_;
+    static constexpr QudaReconstructType recon = recon_;
+    using Link = Matrix<complex<real>, nColor>;
+    static_assert(nColor == 3, "Only nColor=3 enabled at this time");
+    typedef typename gauge_mapper<real,recon>::type Gauge;
+
+    const Gauge u;
+
+    const double factor;
```

Comment:
mayeb a comment what factor is used for?

Reply:
https://github.com/lattice/quda/pull/1308/commits/ccb7dd9dfe1dbb485f0d2f05456c9276a79fbafb